pkg/pod/creds_init.go

```diff
@@ -55,7 +55,10 @@ func credsInit(credsImage string, serviceAccountName, namespace string, kubeclie
 		if err != nil {
 			return nil, nil, err
 		}
-
+		// if there's a github app annotation on the secret let's match that to the github org this pipeline is for
+		if owner != "" && secret.Annotations[gitHubAppAnnotationKey] != "" && secret.Annotations[gitHubAppAnnotationKey] != owner {
```

Comment:
If there's a github app annotation on the secret but no gh org on the pipeline, should the secret still be mounted?  It looks like it would be at the moment.